src/create.ts

```diff
@@ -42,33 +65,30 @@ export function create<
    */
   function useMappedState<TResult>(
     mapState: (state: TState) => TResult,
+    shouldMemoize: Boolean = true,
   ): TResult {
     const store = useContext(StoreContext);
     if (!store) {
       throw new MissingProviderError();
     }
-    const runMapState = () => mapState(store.getState());
-
-    const [derivedState, setDerivedState] = useState(runMapState);
-
-    const lastStore = useRef(store);
-    const lastMapState = useRef(mapState);
-
-    const wrappedSetDerivedState = () => {
-      const newDerivedState = runMapState();
-      setDerivedState(lastDerivedState =>
-        shallowEqual(newDerivedState, lastDerivedState)
-          ? lastDerivedState
-          : newDerivedState,
-      );
-    };
-
-    // If the store or mapState change, rerun mapState
-    if (lastStore.current !== store || lastMapState.current !== mapState) {
-      lastStore.current = store;
-      lastMapState.current = mapState;
-      wrappedSetDerivedState();
-    }
+
+    const memoizedMapState = useMemo(() => {
+      if (shouldMemoize) {
+        return memoize(mapState);
```

Comment:
this is going to have almost no overhead. Any reason not to memoize it unconditionally? I'd like to avoid adding options to `useMappedState`

Reply:
Not really. Removed it.